langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchOrganicResult.java

```diff
@@ -0,0 +1,221 @@
+     * @param url             The URL associated with the search result.
+     * @param snippet         The snippet of the search result, in plain text.
+     * @param content The most query related content from the scraped url.
+     * @param metadata  The metadata associated with the search result.
+     */
+    public WebSearchOrganicResult(String title, URI url, String snippet, String content, Map<String, String> metadata) {
+        this.title = ensureNotBlank(title, "title");
+        this.url = ensureNotNull(url,"url");
+        this.snippet = snippet;
+        this.content = content;
+        this.metadata = getOrDefault(metadata, new HashMap<>());
+    }
+
+    /**
+     * Returns the title of the web page.
+     *
+     * @return The title of the web page.
+     */
+    public String title() {
+        return title;
+    }
+
+    /**
+     * Returns the URL associated with the web page.
+     *
+     * @return The URL associated with the web page.
+     */
+    public URI url() {
+        return url;
+    }
+
+    /**
+     * Returns the snippet associated of the web page.
+     *
+     * @return The snippet associated of the web page.
+     */
+    public String snippet() {
+        return snippet;
+    }
+
+    /**
+     * Returns the content related from the scraped url.
+     *
+     * @return The content related from the scraped url.
+     */
+    public String content() {
+        return content;
+    }
+
+    /**
+     * Returns the result metadata associated with the search result.
+     *
+     * @return The result metadata associated with the search result.
+     */
+    public Map<String, String> metadata() {
+        return metadata;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        WebSearchOrganicResult that = (WebSearchOrganicResult) o;
+        return Objects.equals(title, that.title)
+                && Objects.equals(url, that.url)
+                && Objects.equals(snippet, that.snippet)
+                && Objects.equals(content, that.content)
+                && Objects.equals(metadata, that.metadata);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(title, url, snippet, content, metadata);
+    }
+
+    @Override
+    public String toString() {
+        return "WebSearchOrganicResult{" +
+                "title='" + title + '\'' +
+                ", url=" + url +
+                ", snippet='" + snippet + '\'' +
+                ", content='" + content + '\'' +
+                ", metadata=" + metadata +
+                '}';
+    }
+
+    /**
+     * Converts this WebSearchOrganicResult to a TextSegment.
+     *
+     * @return The TextSegment representation of this WebSearchOrganicResult.
+     */
+    public TextSegment toTextSegment() {
+        return TextSegment.from(content != null ? content : snippet,
```

Comment:
just to double-check: can both be not null at the same time?

Reply:
Yes, that is possible. If both be not null, my logic is that `content` prepend to `snippet` bcz it's supposed that `content` represent  web page scrapped by an AI (e.g Tavily Search have a closed AI to do that) or enriched by an external AI (a new LM integration) using embeding model, store, splitter, etc (v2).